openapi-docs/docs/DefaultApi.md

```diff
@@ -0,0 +1,224 @@
+
+### Return type
+
+null (empty response body)
+
+### Authorization
+
+No authorization required
+
+### HTTP request headers
+
+- **Content-Type**: Not defined
+- **Accept**: Not defined
+
+
+## subscribe
+
+> subscribe(_number, cvc, expMonth, expYear, email, accessControlAllowOrigin, accessControlAllowHeaders, contentType)
+
+subscribe
+
+Create a new stripe user and a subscription
+
+### Example
+
+```javascript
+import AmbianicFunctionsCollection from 'ambianic_functions_collection';
+
+let apiInstance = new AmbianicFunctionsCollection.DefaultApi();
+let _number = CREDIT_CARD_NUMBER; // String | 16 digit Number on the credit card supplied
+let cvc = CREDIT_CARD_CVC; // String | Cvc number behind credit card
+let expMonth = CREDIT_CARD_EXPIRY_MONTH; // String | 2 digit expected credit card expiry month. 
+let expYear = CREDIT_CARD_EXPIRY_YEAR; // String | 4 digit expected credit card expiry year. 
+let email = SUBSCRIBER_EMAIL; // String | Subscriber's email address
+let accessControlAllowOrigin = *; // String | 
+let accessControlAllowHeaders = Content-Type; // String | 
+let contentType = application/json; // String | 
+apiInstance.subscribe(_number, cvc, expMonth, expYear, email, accessControlAllowOrigin, accessControlAllowHeaders, contentType, (error, data, response) => {
+  if (error) {
+    console.error(error);
+  } else {
+    console.log('API called successfully.');
+  }
+});
+```
+
+### Parameters
+
+
+Name | Type | Description  | Notes
+------------- | ------------- | ------------- | -------------
+ **_number** | **String**| 16 digit Number on the credit card supplied | 
+ **cvc** | **String**| Cvc number behind credit card | 
+ **expMonth** | **String**| 2 digit expected credit card expiry month.  | 
+ **expYear** | **String**| 4 digit expected credit card expiry year.  | 
+ **email** | **String**| Subscriber&#39;s email address | 
+ **accessControlAllowOrigin** | **String**|  | 
+ **accessControlAllowHeaders** | **String**|  | 
+ **contentType** | **String**|  | 
+
+### Return type
+
+null (empty response body)
+
+### Authorization
+
+No authorization required
+
+### HTTP request headers
+
+- **Content-Type**: application/x-www-form-urlencoded
+- **Accept**: Not defined
+
+
+## subscription
+
+> subscription(subscriptionId, accessControlAllowOrigin, accessControlAllowHeaders, contentType)
+
+subscription
+
+Delete a user&#39;s subscription on Stripe, either active or expired.
+
+### Example
+
+```javascript
+import AmbianicFunctionsCollection from 'ambianic_functions_collection';
+
+let apiInstance = new AmbianicFunctionsCollection.DefaultApi();
+let subscriptionId = SUBSCRIBER-ID; // String | ID for the user's subscription to be deleted
+let accessControlAllowOrigin = *; // String | 
+let accessControlAllowHeaders = Content-Type; // String | 
+let contentType = application/json; // String | 
+apiInstance.subscription(subscriptionId, accessControlAllowOrigin, accessControlAllowHeaders, contentType, (error, data, response) => {
```

Comment:
@vickywane JS code should look like `apiInstance.cancelSubscription()` or `apiInstance.unsubscribe()` or `apiInstance.deleteSubscription()` . Either of these option is acceptable. Whereas `apiInstance.subscription()` is a terrible name for a JS function that deletes cancels user subscription. Once again, this comes to the difference in naming REST API endpoints vs their corresponding JS functions via the `operationId` parameter in the API spec document.

BTW, where is the API spec file? Looks like its gone from this PR. It should be the single source of truth where API interfaces and docs are generated from, right?

Reply:
The API spec file is the YAML in the root directory with an auto-generated name.  It has now been renamed to `openapi-spec.yaml`. You can review it in the root directory

Reply:
The naming recommendations above have also been implemented.